src/rgw/rgw_sync_module_pubsub.cc

```diff
@@ -753,7 +710,42 @@ class PSSubscription {
     }
   };
 
-  using PushCR = RGWPostRESTResourceCR<rgw_pubsub_event, int>;
+  // Async execution of RGWPostHTTPData via coroutine
+  class RGWPostHTTPDataCR : public RGWPostHTTPData, public RGWCoroutine {
+    private:
+      RGWDataSyncEnv *sync_env;
+      bufferlist read_bl;
+
+    public:
+      RGWPostHTTPDataCR(const std::string& _post_data,
+                    RGWDataSyncEnv* _sync_env,
+                    const string& method,
+                    const string& url,
+                    const bool verify_ssl,
+                    const header_spec_t intercept_headers = {}) :
+      RGWPostHTTPData(_sync_env->cct, method, url, &read_bl, verify_ssl, intercept_headers),
+      RGWCoroutine(_sync_env->cct), sync_env(_sync_env)
+      {
+        set_post_data(_post_data);
+        set_send_length(_post_data.length());
+      }
+
+    int operate() override {
+      reenter(this) {
+        // send message to endpoint and wait for answer
+        auto rc = sync_env->http_manager->add_request(this);
+        if (rc >= 0) {
+          // wait for answer
+          rc = RGWPostHTTPData::wait();
```

Comment:
@yuvalif this can't be right. Should not wait() inside a coroutine. Probably need to create a new coroutine that spawns this asynchronously. See examples in `rgw_cr_rados.{h.cc}`, and `rgw_cr_rest.{h,cc}`

Reply:
@yehudasa isn't that what "yield" at line 798 does?

Reply:
@yuvalif the yield will give the context to another cr once the block finishes running, but if it is blocked, no one will run.